packages/@aws-cdk-testing/framework-integ/test/core/test/integ.nested-stack-references.ts

```diff
@@ -0,0 +1,32 @@
+import * as cdk from 'aws-cdk-lib/core';
+import * as integ from '@aws-cdk/integ-tests-alpha';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as s3 from 'aws-cdk-lib/aws-s3';
+
+const app = new cdk.App();
+const producer = new cdk.Stack(app, 'Producer');
+const nested = new cdk.NestedStack(producer, 'Nestor');
+
+const vpc = new ec2.Vpc(nested, 'MyVpc', { maxAzs: 1 });
+const vpce = new ec2.InterfaceVpcEndpoint(nested, 'MyVpcEndpoint', {
+  vpc,
+  service: {
+    name: `com.amazonaws.${nested.region}.execute-api`,
+    port: 443,
+  },
+});
+
+const consumer = new cdk.Stack(app, 'Consumer');
+new s3.Bucket(consumer, 'ConsumerBucket');
+
+new cdk.CfnOutput(consumer, 'StringAttribute', {
+  value: vpce.vpcEndpointId,
+});
+
+new cdk.CfnOutput(consumer, 'ListAttribute', {
+  value: cdk.Fn.join('$$', vpce.vpcEndpointDnsEntries),
+});
```

Comment:
Running an assertion to make sure the values are as expected would be nice, something like:

https://github.com/aws/aws-cdk/blob/d4f6946dbc97161749a33d3e89514a8f7ca4aa0a/packages/@aws-cdk-testing/framework-integ/test/aws-codeguruprofiler/test/integ.profiler-group-import-functions.ts#L59-L65

Reply:
Added

Reply:
Yeah that's definitely a problem, I can reproduce it in another integration test. I'll investigate further and open an issue, thanks for letting me know. I'll finish the review once that's done